Source/Scene/ModelInstanceCollection.js

```diff
@@ -0,0 +1,425 @@
+/*global define*/
+define([
+        '../Core/BoundingSphere',
+        '../Core/Cartesian3',
+        '../Core/clone',
+        '../Core/Color',
+        '../Core/ComponentDatatype',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/DeveloperError',
+        '../Core/destroyObject',
+        '../Core/Matrix4',
+        '../Renderer/Buffer',
+        '../Renderer/DrawCommand',
+        '../Renderer/WebGLConstants',
+        '../ThirdParty/when',
+        './Model',
+        './SceneMode'
+    ], function(
+        BoundingSphere,
+        Cartesian3,
+        clone,
+        Color,
+        ComponentDatatype,
+        defaultValue,
+        defined,
+        defineProperties,
+        DeveloperError,
+        destroyObject,
+        Matrix4,
+        Buffer,
+        DrawCommand,
+        WebGLConstants,
+        when,
+        Model,
+        SceneMode) {
+    "use strict";
+
+    var LoadState = {
+        NEEDS_LOAD : 0,
+        LOADING : 1,
+        LOADED : 2,
+        FAILED : 3
+    };
+
+    /**
+     * A 3D model instance collection. All instances reference the same underlying model, but have unique
```

Comment:
Let's mark this `@private` until we hook it up to 3D Tiles.  This way we can change it if needed.  We probably won't merge it until then either, but we can still merge billboard instancing.